resolver/query_logging_resolver.go

```diff
@@ -125,6 +144,44 @@ func (r *QueryLoggingResolver) Resolve(request *model.Request) (*model.Response,
 	return resp, err
 }
 
+func (r *QueryLoggingResolver) createLogEntry(request *model.Request, response *model.Response,
+	start time.Time, durationMs int64) *querylog.LogEntry {
+	entry := querylog.LogEntry{
+		Start:       start,
+		ClientIP:    "0.0.0.0",
+		ClientNames: []string{"none"},
+	}
+
+	if slices.Contains(r.fields, config.QueryLogFieldClientIP) {
+		entry.ClientIP = request.ClientIP.String()
+	}
+
+	if slices.Contains(r.fields, config.QueryLogFieldClientName) {
+		entry.ClientNames = request.ClientNames
+	}
+
+	if slices.Contains(r.fields, config.QueryLogFieldResponseReason) {
+		entry.ResponseReason = response.Reason
+		entry.ResponseType = response.RType.String()
+		entry.ResponseCode = dns.RcodeToString[response.Res.Rcode]
+	}
+
+	if slices.Contains(r.fields, config.QueryLogFieldResponseAnswer) {
+		entry.Answer = util.AnswerToString(response.Res.Answer)
+	}
+
+	if slices.Contains(r.fields, config.QueryLogFieldQuestion) {
+		entry.QuestionName = request.Req.Question[0].Name
+		entry.QuestionType = dns.TypeToString[request.Req.Question[0].Qtype]
+	}
+
+	if slices.Contains(r.fields, config.QueryLogFieldDuration) {
+		entry.DurationMs = durationMs
+	}
+
+	return &entry
+}
```

Comment:
Since we're doing this for every query, and with multiple fields, I think it would make sense to refactor this to not iterate over `r.fields` a bunch of times via `slices.Contains`.

A couple possibilities come to mind:

1. use a set `map[config.QueryLogField]struct{}`/`map[config.QueryLogField]bool`:   
   `if _, ok := r.fields[config.QueryLogFieldClientIP]; ok { ... }`

2. iterate over `r.fields` and have a `switch` that sets the correct info on `entry`

Reply:
Yes, that makes sense 👍 will rework this function